### DISCUSSION TITLE
Is there any support for custom post types?

### DISCUSSION THREAD

**Discussion by KeriGerheart** [colbyfayock/next-wordpress-starter#467] [General]
Wondering if there is something already built into this starter to handle WP custom post types? If not, are they are resources someone could point me to? Thanks a lot, this is a really cool project!

**Comment by colbyfayock** [comment]
hey @KeriGerheart custom post types should really "just work" with WPGraphQL though they wouldn't work out of the box with this project

what you can do is basically clone the data and lib files for posts and set up the queries to request that custom post type
- https://github.com/colbyfayock/next-wordpress-starter/blob/main/src/lib/posts.js
- https://github.com/colbyfayock/next-wordpress-starter/blob/main/src/data/posts.js

here's an example of what i've done on my website spacejelly.dev which was based on this starter:
- https://github.com/colbyfayock/spacejelly.dev/blob/main/src/lib/courses.js
- https://github.com/colbyfayock/spacejelly.dev/blob/main/src/data/courses.js

I use the Custom Post Type UI plugin

https://wordpress.org/plugins/custom-post-type-ui/

<img width="1235" alt="image" src="https://user-images.githubusercontent.com/1045274/228026254-3c847ea5-626d-4283-a90e-4fabfdda7eee.png">

<img width="1557" alt="image" src="https://user-images.githubusercontent.com/1045274/228026572-faaf0a47-8f1f-4ffd-8cd2-334f0178c5e4.png">

**Comment by KeriGerheart** [comment]
Awesome, thanks very much @colbyfayock! Great project you made here, really appreciate you sharing all your hard work and knowledge!
